api/filtering/filtering.py

```diff
@@ -270,6 +272,20 @@ def _generic_filter_builder(builder_function, field_name, field_value, field_fil
         return _base_filter_builder(nullable_builder_function, field_name, field_value, field_filter, operation, spec)
 
 
+def _build_registered_with_per_reporter_filter(registered_with):
+    prs_list = []
+    for item in registered_with:
+        prs_list.append(
```

Comment:
I just realized that we'll need to keep handling the `registered_with=insights` case so that we don't break its current functionality. With the refactoring you've done, though, this should be a pretty simple change:
```suggestion
    for item in registered_with:
        if item == "insights":
            prs_list.append({"NOT": {"insights_id": {"eq": None}}})
        else:
            prs_list.append(
```